data-runtime/src/main/java/io/micronaut/data/runtime/query/internal/DefaultStoredQuery.java

```diff
@@ -116,8 +109,11 @@ public DefaultStoredQuery(
                 method.stringValue(DATA_METHOD_ANN_NAME, TypeRole.SORT).isPresent() ||
                 method.intValue(DATA_METHOD_ANN_NAME, META_MEMBER_LIMIT).orElse(-1) > -1 ||
                 method.intValue(DATA_METHOD_ANN_NAME, META_MEMBER_PAGE_SIZE).orElse(-1) > -1;
-
+        String query;
         if (isCount) {
+            query = method.stringValue(Query.class, DataMethod.META_MEMBER_COUNT_QUERY)
+                .orElseGet(() -> method.stringValue(Query.class)
+                    .orElseThrow(() -> new IllegalStateException("No query present in method")));
```

Comment:
Maybe error message to contain `count` to be easier to figure out what was wrong? `"No count query present in method"`